lib/Graph/Node.cpp

```diff
@@ -53,6 +53,9 @@ void NodeValue::replaceAllUsesOfWith(NodeValue v) {
                                          nodeValueUsers.end());
   for (auto &U : usersVec) {
     NodeHandle *site = U.get();
+    auto *userF = U.getUser()->getParent();
```

Comment:
Could you please add a comment that says something like:  "If the user is not in function F then don't touch it".